scripts/push_pkg.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/sh
+
+set -e
+
+topdir="$(realpath "$(dirname "${0}")/../openwrt")"
+
+# defaults to qemu run script
+ssh_port=2223
+ssh_host=localhost
+build_only=0
+
+print_help() {
+	echo "$0 [OPTIONS] PACAKGE_DIR"
+	echo ""
+	echo " -h          print this help"
+	echo " -r HOST     use ssh host (default port will be 22)"
+	echo " -p PORT     use ssh port"
+	echo " -b          build only, do not push anywhere"
+	echo ""
+	echo ' To change gluon variables, run e.g. "make config GLUON_MINIFY=0"'
+	echo ' because then the gluon logic will be triggered, and openwrt/.config'
+	echo ' will be regenerated. The variables from openwrt/.config are already'
+	echo ' automatically used for this script.'
+	echo
+}
+
+while getopts "p:r:hb" opt
+do
+	case $opt in
+		p) ssh_port="${OPTARG}";;
+		r) ssh_host="${OPTARG}"; ssh_port=22;;
+		b) build_only=1;;
+		h) print_help; exit 0;;
+		*) ;;
+	esac
+done
+shift $(( OPTIND - 1 ))
+
+if [ "$build_only" -eq 0 ]; then
+	OPENWRT_BOARD="$(ssh -p "${ssh_port}" "root@${ssh_host}" sh -c \''. /etc/os-release; echo "$OPENWRT_BOARD"'\')"
+	OPENWRT_ARCH="$(ssh -p "${ssh_port}" "root@${ssh_host}" sh -c \''. /etc/os-release; echo "$OPENWRT_ARCH"'\')"
+
+	# check target
+	if ! grep "CONFIG_TARGET_ARCH_PACKAGES=\"${OPENWRT_ARCH}\"" "${topdir}/.config" > /dev/null; then
+		echo "Configured OpenWrt Target is not matching remote!" 1>&2
+		printf "%s" "Local: " 1>&2
+		grep "CONFIG_TARGET_ARCH_PACKAGES" "${topdir}/.config" 1>&2
+		echo "Remote: ${OPENWRT_ARCH}" 1>&2
+		echo 1>&2
+		echo "To switch the local with the run with the corresponding GLUON_TARGET:"  1>&2
+		echo "  make GLUON_TARGET=... config" 1>&2
+		exit 1
+	fi
+fi
+
+if [ $# -lt 1 ]; then
+	echo ERROR: Please specify a PACKAGE_DIR. For example:
+	echo
+	echo " \$ $0 package/gluon-core"
+	exit 1
+fi
+
+while [ $# -gt 0 ]; do
+
+	pkgdir="$1"; shift
+	echo "Package: ${pkgdir}"
+
+	if ! [ -f "${pkgdir}/Makefile" ]; then
+		echo "ERROR: ${pkgdir} does not contain a Makefile"
+		exit 1
+	fi
+
+	if ! grep BuildPackage "${pkgdir}/Makefile" > /dev/null; then
+		echo "ERROR: ${pkgdir}/Makefile does not contain a BuildPackage command"
+		exit 1
+	fi
+
+	opkg_packages="$(grep BuildPackage "${pkgdir}/Makefile" | cut -d',' -f 2 | tr -d ')' | tr -d '\n')"
```

Comment:
A better way find out what packages a Makefile contains would be:
```sh
make TOPDIR="${topdir}" -C "${pkgdir}" DUMP=1 | awk '/^Package: / { print $2 }'
```
The DUMP feature is also what OpenWrt uses internally for enumerate packages.